process/coordinator/process.go

```diff
@@ -748,6 +749,48 @@ func (tc *transactionCoordinator) CreateMbsAndProcessCrossShardTransactionsDstMe
 	return createMBDestMeExecutionInfo.miniBlocks, createMBDestMeExecutionInfo.numTxAdded, allMBsProcessed, nil
 }
 
+func (tc *transactionCoordinator) requestMissingMiniBlocksAndTransactions(mbsInfo []*data.MiniBlockInfo) {
+	mapMissingMiniBlocksPerShard := make(map[uint32][][]byte)
+
+	tc.requestedItemsHandler.Sweep()
+
+	for _, mbInfo := range mbsInfo {
+		object, isMiniBlockFound := tc.miniBlockPool.Peek(mbInfo.Hash)
+		if !isMiniBlockFound {
+			log.Debug("transactionCoordinator.requestMissingMiniBlocksAndTransactions: mini block not found and was requested",
+				"sender shard", mbInfo.SenderShardID,
+				"hash", mbInfo.Hash,
+				"round", mbInfo.Round,
+			)
+			mapMissingMiniBlocksPerShard[mbInfo.SenderShardID] = append(mapMissingMiniBlocksPerShard[mbInfo.SenderShardID], mbInfo.Hash)
+			_ = tc.requestedItemsHandler.Add(string(mbInfo.Hash))
+			continue
+		}
+
+		miniBlock, isMiniBlock := object.(*block.MiniBlock)
+		if !isMiniBlock {
+			log.Warn("transactionCoordinator.requestMissingMiniBlocksAndTransactions", "mb hash", mbInfo.Hash, "error", process.ErrWrongTypeAssertion)
+			continue
+		}
+
+		preproc := tc.getPreProcessor(miniBlock.Type)
+		if check.IfNil(preproc) {
+			log.Warn("transactionCoordinator.requestMissingMiniBlocksAndTransactions: getPreProcessor", "mb type", miniBlock.Type, "error", process.ErrNilPreProcessor)
+			continue
+		}
+
+		numTxsRequested := preproc.RequestTransactionsForMiniBlock(miniBlock)
```

Comment:
this will cause the requesting of the missing transactions to be done before the miniblocks are actually requested. This will work as the request handler will whitelist the required data and the interceptors will then accept the missing txs. We should prevent the future disruption of the whitelist mechanisms between the request handler and the interceptors.

Reply:
Actually this is the case when some of the mini blocks, from the same header, are already in pool and it requests txs from them, if they are missing.

Reply:
right